Lib/pydoc.py

```diff
@@ -1775,10 +1776,15 @@ def render_doc(thing, title='Python Library Documentation: %s', forceload=0,
     return title % desc + '\n\n' + renderer.document(object, name)
 
 def doc(thing, title='Python Library Documentation: %s', forceload=0,
-        output=None):
+        output=None, is_cli=False):
     """Display text documentation, given an object or a path to an object."""
     if output is None:
-        pager(render_doc(thing, title, forceload))
+        try:
+            pager(render_doc(thing, title, forceload))
+        except ImportError:
+            if is_cli:
+                raise ImportError(missing_pattern % thing) from None
+            pager(missing_pattern % thing)
```

Comment:
Instead of invoking `pager` screen just to show message about missing `thing`, you can print this message directly to stdout, just as 3.10 does.
```python
print(missing_pattern % thing)
```